### DISCUSSION TITLE
My PSU is fried 🥳🥳 (+vast.ai guide)

### DISCUSSION THREAD

**Discussion by oobabooga** [oobabooga/text-generation-webui#528] [Announcements]
It happened while I was trying to quantize a model. It should take another week for a replacement to arrive, so for the past few days I have been renting random GPUs from [vast.ai](https://cloud.vast.ai/?ref=61324) for testing updates.

It's really easy to set up the web UI there: just select the `pytorch:latest` image on the left and select `Run interactive shell server, SSH` as the launch mode. Once your chosen machine boots, SSH into it and copy and paste these commands to get everything running (including 8-bit and 4-bit modes):

```
apt install -y build-essential vim
git clone https://github.com/oobabooga/text-generation-webui
cd text-generation-webui
pip install -r requirements.txt
cp $(ls /opt/conda/lib/python3.10/site-packages/bitsandbytes/*cuda*.so | grep -v nocublaslt | tail -1) /opt/conda/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cpu.so
conda install -y cudatoolkit
mkdir repositories
cd repositories
git clone https://github.com/qwopqwop200/GPTQ-for-LLaMa -b cuda
cd GPTQ-for-LLaMa
python setup_cuda.py install
cd ../..
echo 'export LD_LIBRARY_PATH=/opt/conda/lib:$LD_LIBRARY_PATH' >> ~/.bashrc
export LD_LIBRARY_PATH=/opt/conda/lib:$LD_LIBRARY_PATH
```

If you want to download llama-7b, these are the commands:

```
python download-model.py decapoda-research/llama-7b-hf
sed -i 's/LLaMATokenizer/LlamaTokenizer/g' models/llama-7b-hf/tokenizer_config.json
cd models
wget 'https://huggingface.co/decapoda-research/llama-7b-hf-int4/resolve/main/llama-7b-4bit.pt'
cd ..
```

I find that 60GB of storage is more than enough for everything. You can probably get away with just 20GB if you are only interested in a single model. Also, if you hover over the "Rent" button, it tells you the internet price per GB for that instance. Try picking an instance that doesn't charge anything for downloads.

Make sure to launch the web UI with the `--share` flag to generate a public gradio URL.

If you want to try vast.ai, use this link to help me gain some credits (I promise this is not some elaborate grift, their service really is useful): https://cloud.vast.ai/?ref=61324

=============

**UPDATE**: I got a new one, all is back to normal now.

**Comment by RandomInternetPreson** [comment]
:c oh nos!!! your PSU died! frick that would scare the crap out of me, especially if it happened with the gpu under load like that.  Thank you for everything <3

**Comment by Snowad14** [comment]
Now that I have a 3090, I can share some of my Vast.ai tips:
- Avoid using pytorch:latest because it's based on an older version of cuda (10.1), take rather the second one which is based on more recent versions like 11.7 with pytorch 2.0
- The most important : check "Unverified Machines" and you can easily find machines 2 to 3x cheaper ! (~1.2/hr to 0.90/hr $ on good offers for a 3090)
- This advice comes with the one above, check the download speed but especially their cost (visible when hovering over the "RENT" button) because if the price of the download is too high or that the connection is too slow and you want to download large models it can be expensive (so you have to find the right balance between the cost of the download and the cost of $/h)

**Reply by Teragron** [reply]
I'm getting the following error in one of the unverified 4090 GPU in vastai:                                                                                        `ImportError: /opt/conda/lib/python3.10/site-packages/torch/lib/libtorch_cuda.so: symbol cudaGraphInstantiateWithFlags version libcudart.so.11.0 not defined in file libcudart.so.11.0 with link time reference`                                                                                                        This was my version tag: pytorch/pytorch:2.0.0-cuda11.7-cudnn8-runtime.                                                                                                    Did anyone experience something similar?

**Reply by Snowad14** [reply]
> I'm getting the following error in one of the unverified 4090 GPU in vastai: `ImportError: /opt/conda/lib/python3.10/site-packages/torch/lib/libtorch_cuda.so: symbol cudaGraphInstantiateWithFlags version libcudart.so.11.0 not defined in file libcudart.so.11.0 with link time reference` This was my version tag: pytorch/pytorch:2.0.0-cuda11.7-cudnn8-runtime. Did anyone experience something similar?

There are machines that don't work, just get another one

**Comment by sanskar-mk2** [comment]
Here are some solutions that helped me when I run into error when pip installing the requirements: #1534
